code/modules/mob/mob_helpers.dm

```diff
@@ -361,9 +361,16 @@ var/list/intents = list(I_HELP,I_DISARM,I_GRAB,I_HARM)
 	set name = "Rest"
 	set category = "IC"
 
-	resting = !resting
-	update_canmove()
-	to_chat(src, "<span class='notice'>You are now [resting ? "resting" : "getting up"].</span>")
+	if(world.time < client.move_delay)
+		return
+
+	if(!resting)
+		client.move_delay += 20

```

Comment:
Pretty sure this should be
```
client.move_delay = world.time + 20
```
so you can't "save up" a difference to flop like a fish for a while.